src/modules/Dashboard/components/TakeNowModal/LabeledDropdown/LabeledUserDropdown.tsx

```diff
@@ -64,42 +71,105 @@ export const LabeledUserDropdown = ({
     [debounce, handleSearch],
   );
 
+  const shouldShowWarningMessage = !!canShowWarningMessage && !!value && value.tag !== 'Team';
+
+  let groupBy: LabeledUserDropdownProps['groupBy'];
+
+  if (showGroups) {
+    groupBy = (option: ParticipantDropdownOption) => {
+      if (option.tag !== 'Team') {
+        return t('takeNow.modal.dropdown.participantGrouping');
+      } else {
+        return t('takeNow.modal.dropdown.teamMemberGrouping');
+      }
+    };
+  }
+
   return (
-    <StyledFlexColumn sx={{ gap: 1.6, marginTop: 2.4 }}>
+    <StyledFlexColumn sx={{ gap: 1.6, ...sx }}>
       <Box sx={{ display: 'flex', gap: 0.4 }}>
-        <StyledTitleMedium fontWeight="bold">{label}</StyledTitleMedium>
-        <Tooltip tooltipTitle={tooltip}>
-          <Box sx={{ height: 24 }}>
-            <StyledTitleTooltipIcon
-              sx={{ marginLeft: 0 }}
-              id="more-info-outlined"
-              width={24}
-              height={24}
-              data-testid={`${rest['data-testid']}-tooltip-icon`}
-            />
-          </Box>
-        </Tooltip>
+        <StyledTitleMedium sx={{ fontWeight: 700, color: variables.palette.on_surface }}>
+          {label}
+        </StyledTitleMedium>
+        <RenderIf condition={!!tooltip}>
+          <Tooltip tooltipTitle={tooltip}>
```

Comment:
Yeah the only use case I created it for was avoiding the conditionals inside the JSX. I find logic + markup hard to read, especially when there's a lot of it. This is a good point I hadn't considered - `MyComponent` is still a function that gets called, even if its output doesn't get used.

I could change the definition of `RenderIf` to something like this

```typescript
/**
 * Render a component tree if a condition is met, otherwise render a fallback component. By default, the fallback is null.
 */
export const RenderIf = <T extends React.ComponentType<any>>({
  condition,
  Component,
  fallback,
  ...props
}: RenderIfProps<T> & React.ComponentProps<T>) => {
  if (condition) {
    return <Component {...props} />;
  }

  return fallback?.() ?? null;
};
```

So that its usage would now look like this

```typescript
<RenderIf condition={!!title} Component={MyComponent} title={title!} />
```

This would avoid the unnecessary evaluation of `MyComponent`, but we still have to assert that the title is not undefined. I don't love it, but I think it's an improvement. What do you think?

Reply:
I'll keep working on it. For now I'll fall back to a conditional